src/inputoutput.jl

```diff
@@ -239,16 +240,22 @@ function generate_control_function(sys::AbstractODESystem, inputs = unbound_inpu
     t = get_iv(sys)
 
     # pre = has_difference ? (ex -> ex) : get_postprocess_fbody(sys)
-
-    args = (u, inputs, p..., t)
+    if disturbance_argument
+        args = (u, inputs, p..., t, disturbance_inputs)
+    else
+        args = (u, inputs, p..., t)
+    end
     if implicit_dae
         ddvs = map(Differential(get_iv(sys)), dvs)
         args = (ddvs, args...)
     end
     process = get_postprocess_fbody(sys)
+    wrapped_arrays_vars = disturbance_argument ?
+                          wrap_array_vars(sys, rhss; dvs, ps, inputs, disturbance_inputs) :
+                          wrap_array_vars(sys, rhss; dvs, ps, inputs)
```

Comment:
```suggestion
    wrapped_arrays_vars = disturbance_argument ?
                          wrap_array_vars(sys, rhss; dvs, ps, inputs, cachesyms = (disturbance_inputs,)) :
                          wrap_array_vars(sys, rhss; dvs, ps, inputs)
```

`wrap_array_vars` is basically a workaround in our codegen to handle cases where an array variable is scalarized in unknowns/parameters but used in the unscalarized form in equations. The best solution is to pass `disturbance_inputs` to `cachesyms` as a tuple. This only works because `w` is the last argument to the generated function. I'll add documentation to `wrap_array_vars` in another PR, but it's _very_ dependent on what the arguments to the generated function are. As a result, it carries some bad API baggage to avoid having to refactor the million places it's called. Once I find the time, I plan to overhaul our codegen so it isn't implemented in such a scattered manner and has some new optimizations. That will also get rid of this API baggage and hopefully make `wrap_array_vars` more concise.